x-pack/plugins/enterprise_search/public/applications/shared/deprecation_callout/deprecation_callout.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiCallOut, EuiButton, EuiLink, EuiFlexItem, EuiFlexGroup, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface DeprecationCalloutProps {
+  onDismissAction: () => void;
+}
+
+export const EnterpriseSearchDeprecationCallout: React.FC<DeprecationCalloutProps> = ({
+  onDismissAction,
+}) => {
+  return (
+    <EuiCallOut
+      onDismiss={onDismissAction}
+      iconType={'warning'}
+      color={'warning'}
+      title={i18n.translate(
+        'xpack.enterpriseSearch.enterpriseSearchDeprecationCallout.euiCallOut.title',
+        { defaultMessage: 'Important Note' }
+      )}
+      data-test-subj="EnterpriseSearchDeprecationCallout"
+    >
+      <FormattedMessage
+        id="xpack.enterpriseSearch.deprecationCallout.first_message"
+        defaultMessage="The standalone App Search and Workplace Search products remain available in maintenance mode, and are not recommended for new search experiences. Instead, we recommend using our Elasticsearch-native tools which we are actively developing and improving, for your search use cases. These tools offer the flexibility and composability of working directly with Elasticsearch indices."
+      />
+      <EuiSpacer size="s" />
+      <FormattedMessage
+        id="xpack.enterpriseSearch.deprecationCallout.second_message"
+        defaultMessage="See this {blogUrl} for more information about upgrading your internal knowledge search or this blog post about upgrading your catalog search."
+        values={{
+          blogUrl: (
+            <EuiLink
+              data-test-subj="workplaceSearch-deprecationCallout-blog-link"
+              href="https://www.elastic.co/search-labs/blog/evolution-app-search-elasticsearch"
```

Comment:
should this be in a doc link instead of hard-coded?

Reply:
They could, but I was under the impression those links were tied to actual documentation, and not Search Labs blogs, no? Regardless - where do these links get populated? I can't seem to find it in the code...

Reply:
Thanks again @TattdCodeMonkey - this latest commit should do the trick.